### PR TITLE
Issue/fix supported audio subtypes

### PR DESCRIPTION
**Pull request by marecar3**
Preparing for fix [#8052](https://github.com/wordpress-mobile/WordPress-Android/issues/8052) - Media Library: Allow uploading audio files

There was a bit confusion about file extensions and mime-types so I noticed that `MediaUtils.java` needed to be fixed in order to allow uploading audio files.

When we compare [Wordpress accepted file types](https://en.support.wordpress.com/accepted-filetypes/) with [Android Mime types](https://android.googlesource.com/platform/frameworks/base/+/56a2301/media/java/android/media/MediaFile.java), under the implementation of `MimeTypeMap.getSingleton().getMimeTypeFromExtension()`Android API will have something like this for Audio files types `{"extension" : "mime_type"}`  ->`{"mp3" : "audio/mpeg", "m4a" : "audio/mp4", "ogg" : "audio/ogg", "wav" : "audio/x-wav"}`.

With this fix uploading audio files will work in the example app.

In the future, [Danilo's PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/509) can be used to fix confusion for all other file types (image, video etc..)

Note: After this fix It wasn't easy to align audio file tests with other tests so I needed to tweak some audio file tests to stop failing

Edit : Special case -> On Samsung device for extension "wav" Android API will return "audio/vnd.wave"

### CODE REVIEW

**Comment by daniloercoli** [comment]
I took a look at this PR, and looks good...although  I'm wondering if we need to fix the problems you've mention above, about confusion in file extensions vs. mime-types in FluxC, before preceding further with adding a new feature (#4842) and allow other file types. 

The new code in this PR looks good, but it diverge a bit from the original "bugged" code, and makes other people hard to follow what is a correct mime/type or not. Also variable naming, in the original code, doesn't help on this and make even more confusion.

Note related to this PR:
I also don't understand how FluxC could work fine on uploading a picture to a site, since it seems to me that the mime type set on the request, for example on a `.jpg` picture is `image/jpg`, (see `MediaUtils.java-->getMimeTypeForExtension`) that is not a valid mime type. Maybe WordPress the server does validate the file by looking at the header of the file, before using the one passed in the request.

**Comment by nbradbury** [comment]
@daniloercoli First off, thanks for taking the time to look at this! I definitely agree we need to fix the MIME/extension confusion, but I don't think we need to do that as part of this trial.

@marecar3 Honestly, I wasn't aware FluxC work would be required for this trial, so you went above and beyond by diving into FluxC. For now, you can simply use the hash from this PR to work on the WPAndroid side of things - see [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/build.gradle#L72). 

I think we can decide closer to the end of the trial whether to merge this PR and the related WPAndroid one (although I think we should unless we believe we're breaking something).

File: fluxc/src/main/java/org/wordpress/android/fluxc/utils/MediaUtils.java [hunk 1]
```diff
@@ -27,8 +27,9 @@
     public static final String[] SUPPORTED_VIDEO_SUBTYPES = {
             "mp4", "m4v", "mov", "wmv", "avi", "mpg", "ogv", "3gp", "3gpp", "3gpp2", "3g2", "mpeg", "quicktime", "webm"
     };
+    // supported audio file extensions {"mp3" : "mpeg", "m4a" : "mp4", "ogg" : "ogg", "wav" : "x-wav"}
```
Comment (nbradbury) [inline fluxc/src/main/java/org/wordpress/android/fluxc/utils/MediaUtils.java]: Let's add a comment here explaining why the above is commented out and why we're falling back to a smaller list of audio subtypes.

File: example/src/test/java/org/wordpress/android/fluxc/utils/MediaUtilsTest.java [hunk 2]
```diff
@@ -97,7 +96,7 @@ public void testSupportedVideoRecognition() {
             String supportedVideoMimeType = MediaUtils.MIME_TYPE_VIDEO + supportedVideoType;
             Assert.assertTrue(MediaUtils.isSupportedVideoMimeType(supportedVideoMimeType));
         }
-        for (String unsupportedVideoType : MediaUtils.SUPPORTED_AUDIO_SUBTYPES) {
+        for (String unsupportedVideoType : MediaUtils.SUPPORTED_IMAGE_SUBTYPES) {
```
Comment (nbradbury) [inline example/src/test/java/org/wordpress/android/fluxc/utils/MediaUtilsTest.java]: Nice catch here :)

File: fluxc/src/main/java/org/wordpress/android/fluxc/utils/MediaUtils.java (hunk 1 above)
Comment (marecar3) [reply fluxc/src/main/java/org/wordpress/android/fluxc/utils/MediaUtils.java]: @nbradbury yup I agree, will add some description here.

Comment (marecar3) [reply fluxc/src/main/java/org/wordpress/android/fluxc/utils/MediaUtils.java]: Fixed.

**Comment by oguzkocer** [comment]
@marecar3 It looks like this PR hasn't been updated in the last 1.5 years. If we are not planning to merge this, could we please close it?

**Comment by marecar3** [comment]
Closing in favor of https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1634
